server/service/src/cursor_controller.rs

```diff
@@ -18,6 +41,95 @@ impl CursorController {
         connection: &StorageConnection,
         cursor: u64,
     ) -> Result<(), RepositoryError> {
-        KeyValueStoreRepository::new(connection).set_i32(self.0.clone(), Some(cursor as i32))
+        match &self.0 {
+            CursorType::Standard(key_type) => self.update_standard(connection, key_type, cursor),
+            CursorType::Dynamic(cursor_id) => self.update_dynamic(connection, cursor_id, cursor),
+        }
+    }
+    fn update_standard(
+        &self,
+        connection: &StorageConnection,
+        key_type: &KeyType,
+        cursor: u64,
+    ) -> Result<(), RepositoryError> {
+        KeyValueStoreRepository::new(connection).set_i32(key_type.clone(), Some(cursor as i32))
+    }
+    fn update_dynamic(
+        &self,
+        connection: &StorageConnection,
+        cursor_id: &str,
+        cursor: u64,
+    ) -> Result<(), RepositoryError> {
+        let mut json_text = KeyValueStoreRepository::new(connection)
```

Comment:
I wonder if this should be in a transaction with update lock on postgres?
To avoid potential for two processors to have a race condition with the update?
Feels like low risk for now, happy to make it a follow up issue?

Reply:
Good idea thanks: dc86a2c5871aabcff656332b2865b846698e1576